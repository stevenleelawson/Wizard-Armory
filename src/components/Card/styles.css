.card {
  margin: 1.5em 1em;
}

.card:hover {
  transform: scale(1.1);
}

.isSelected {
  border: 1px solid blue;
}
