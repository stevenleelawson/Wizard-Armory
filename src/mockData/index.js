export const mockCardData = [{
  name: "As Luck Would Have It",
  manaCost: "{G}",
  cmc: 1,
  type: "Enchantment",
  rarity: "Rare",
  set: "UST",
  setName: "Unstable",
  text: "Hexproof Whenever you roll a die, put a number of luck counters on As Luck Would Have It equal to the result. Then if there are 100 or more luck counters on As Luck Would Have It, you win the game. (Count both rolls if you reroll a die.)",
  artist: "Milivoj Ä†eran",
  number: "102",
  layout: "normal",
  multiverseid: 439491,
  imageUrl: "http://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=439491&type=card",
  originalText: "Hexproof Whenever you roll a die, put a number of luck, counters on As Luck Would Have It equal to the result. Then if there are 100 or more luck counters on As Luck Would Have It, you win the game. (Count both rolls if you reroll a die.)",
  originalType: "Enchantment",
  id: "8d2722825dfa475154e0c87ba72c47528c4cf485"
}];
