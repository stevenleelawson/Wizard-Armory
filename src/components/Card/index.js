import React from 'react';
import PropTypes from 'prop-types';
import './styles.css';

const Card = ({imageUrl, id, info, addToDeck}) => {
  console.log('addToDeck', addToDeck)
  const handleDeckBtnClick = () => {
    addToDeck(info)
  }

  return (
    <div className='card' onClick={(event) => handleDeckBtnClick()}>
      <img src={imageUrl} key={id} alt='magic card'/>
    </div>
  );
};

export default Card;

Card.propTypes = {
  imageUrl: PropTypes.string,
  id: PropTypes.number
};
