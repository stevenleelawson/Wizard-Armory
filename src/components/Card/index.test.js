import React from 'react';
import { shallow } from 'enzyme';
import Card from './index';
import * as mockData from '../../mockData'

describe('Card', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Card deck={mockData.mockCardData}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
