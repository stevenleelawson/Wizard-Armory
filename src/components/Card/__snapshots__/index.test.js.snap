// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        alt="magic card"
        src={undefined}
/>,
      "className": "card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "magic card",
        "src": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt="magic card"
          src={undefined}
/>,
        "className": "card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "magic card",
          "src": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
